### DISCUSSION TITLE
Verifying Merkle Patrica Tree data

### DISCUSSION THREAD

**Discussion by amritkumarj** [Consensys/gnark#282] [Q&A]
Hello Everybody,
Amrit here I am a blockchain developer and needed help regarding verifying Merkle Patricia Tree data through gnark. Basically, I wanted this for verifying storage data from the state root in ethereum but due to less experience in creating circuits and less resources available online for this topic but having some problems, I contacted @yelhousni and he instructed me to discuss it here to reach the whole team, if anyone can assist me in this it would be a great help.

**Comment by ThomasPiellard** [comment]
Hi @DhruvJain1122 , sorry for the late answer, so we have an example of a binary Merkle proof verification in `std/accumulator` if you want to see how it works. There is no domain separator between leaves and nodes in this example. The verification circuit is not tied to the actual implementation of the Merkle tree: the proof just contains a root hash, the path as a slice of `frontend.Variable`, and a helper slice of `0`'s and `1`'s, telling the circuit the order of the elements to be hashed (`current_node||current_hash `or `current_hash||current_node`).

If you have more specific questions let us know.
